tally_ho/libs/verify/quarantine_checks.py

```diff
@@ -243,6 +247,72 @@ def pass_sum_of_candidates_votes_validation(result_form):
     return diff <= scaled_tolerance
 
 
+def pass_invalid_ballots_percentage_validation(result_form):
+    """Validate the percentage of invalid ballots.
+
+    If the `result_form` does not have a `reconciliation_form` this will
+    always return True.
+
+    Fails if the percentage of invalid ballots differs by a large margin with
+    the allowed percetage value calculated by mutiplying N% tolerance with
+    ballots inside the ballot box.
+
+    :param result_form: The result form to check.
+    :returns: A boolean of true if passed, otherwise false.
+    """
+    recon_form = result_form.reconciliationform
+
+    if not recon_form:
+        return True
+
+    qc = QuarantineCheck.objects.get(
+        method='pass_invalid_ballots_percentage_validation')
+    invalid_ballots_percantage =\
+        (recon_form.number_invalid_votes / 100) * \
+        recon_form.number_ballots_inside_the_box
+    allowed_invalid_ballots_percantage =\
+        (qc.percentage / 100) * recon_form.number_ballots_inside_the_box
+
+    return invalid_ballots_percantage <= allowed_invalid_ballots_percantage
+
+
+def pass_turnout_percentage_validation(result_form):
+    """Validate the turnout percentage.
+
+    If the `result_form` does not have a `reconciliation_form` this will
+    always return True.
+
+    If the `station` for this `result_form` has an empty `registrants` field
+    this will always return True.
+
+    Fails if the turnout percentage differs by a large margin with
+    the allowed percetage value calculated by mutiplying N% tolerance with
+    total votes/number of ballots used.
+
+    :param result_form: The result form to check.
+    :returns: A boolean of true if passed, otherwise false.
+    """
+    recon_form = result_form.reconciliationform
+
+    if not recon_form:
+        return True
+
+    registrants = result_form.station.registrants if result_form.station\
```

Comment:
can `registrants` be `None` or does it have a default/required value?

Reply:
Yes, it can be `None`, it is not a required field.